azure-devops/create-new-agent-image.ps1

```diff
@@ -0,0 +1,297 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# This script assumes you have installed Azure tools into Powershell by following the instructions
+# at https://docs.microsoft.com/en-us/powershell/azure/install-az-ps?view=azps-3.6.1
+# or are running from Azure Cloud Shell
+#
+
+$Location = 'westus2'
+$Prefix = 'StlBuild-' + (Get-Date -Format 'yyyy-MM-dd')
+$VMSize = 'Standard_D16s_v3'
+$ProtoVMName = 'PROTOTYPE'
+$LiveVMPrefix = 'BUILD'
+$WindowsServerSku = '2019-Datacenter'
+
+$ProgressActivity = 'Creating Scale Set'
+$TotalProgress = 8
+$CurrentProgress = 1
+
+function Find-ResourceGroupName {
+  Param(
+    [string] $prefix
+  )
+
+  $result = $prefix
+  $resources = Get-AzResourceGroup
+  $collision = $false
+  foreach ($resource in $resources) {
+    if ($resource.ResourceGroupName -eq $result) {
+      $collision = $true
+      break
+    }
+  }
+
+  $suffix = 0
+  while ($collision) {
+    $collision = $false
+    $suffix++
+    $result = "$prefix-$suffix"
+    foreach ($resource in $resources) {
+      if ($resource.ResourceGroupName -eq $result) {
+        $collision = $true
+        break
+      }
+    }
+  }

```

Comment:
It looks like this logic could be simplified. Instead of having two phases, have a single phase, and simply select between `$result = $prefix` and `$result = "$prefix-$suffix"` to avoid appending `-0`. (You'll need to change the timing of the increment.)

Reply:
I don't think that's really simpler because it effectively convolutes two loop bodies further. I tried extracting a function instead, do you like that? (After next update of course)